app/controllers/happy_things_controller.rb

```diff
@@ -71,12 +71,16 @@ def happy_thing_params
   end
 
   def create_happy_thing(happy_thing)
-    if happy_thing.save
-      redirect_to root_path, notice: "Yay! 🎉 #{happy_thing.class} was successfully created."
-    else
-      render :new, status: 422
+    respond_to do |format|
+      if happy_thing.save
+        format.html { redirect_to root_path, notice: "Happy Thing was successfully created." }
+        format.json { render json: { status: :created, happy_thing: happy_thing } }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: happy_thing.errors, status: :unprocessable_entity }
+      end
```

Comment:
You can get rig of the parameter in this method, and use `@happy_thing` since is an instance variable.